src/Variable.cpp

```diff
@@ -39,7 +39,15 @@ VariablePtr dot(VariablePtr var, int ndiff)
   }
 }
 
-VariablePtr Variable::duplicate(const std::string & n) const
+Variable::~Variable()
+{
+  if(!isSubvariable())
+  {
+    delete[] memory_;
+  }
+}
```

Comment:
Isn't there potential for deleting variable memory that is still referenced via sub-variables?

Reply:
Not as far as I can see, because all variables who comes from a same variable `x` (via any sequence of derivations and subvariables) share the ref counter of `x`.